blog/views.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.paginator import Paginator
+from django.db.models import Count
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse_lazy
+from django.views import generic
+
+from blog.models import Commentary, Post
+
+
+def index(request: HttpRequest) -> HttpResponse:
+    posts = (
+        Post.objects.annotate(
+            num_comments=Count("comments")
+        ).order_by("-created_time")
+    )
+
+    for post in posts:
+        words = post.content.split()
+        if len(words) > 10:
+            post.content = " ".join(words[:10]) + "..."
```

Comment:
hmmm, can you do it in the request without `for` loop?

Reply:
I changed the function without using `for` loop, but in the current implementation, it is not whole words that are truncated, but parts of words.